gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTestHelper.java

```diff
@@ -238,6 +254,34 @@ public static byte[] writeObject(
     return partition;
   }
 
+  public static Map<String, byte[]> getDecodedMetadata(Map<String, String> metadata) {
+    return metadata.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                entity -> entity.getKey(), entity -> decodeMetadataValues(entity.getValue())));
+  }
+
+  public static byte[] decodeMetadataValues(String value) {
+    try {
+      return BaseEncoding.base64().decode(value);
+    } catch (IllegalArgumentException iae) {
+      logger.atSevere().withCause(iae).log(
+          "Failed to parse base64 encoded attribute value %s - %s", value, iae);
```

Comment:
Should we just propagate exception, as this is a test code and we do not want to read logs to fix tests?

Reply:
Removed.